nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/PutHiveStreaming.java

```diff
@@ -0,0 +1,657 @@
+            .name("retry")
+            .description("A FlowFile is routed to this relationship if the database cannot be updated but attempting the operation again may succeed")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if the database cannot be updated and retrying the operation will also fail.")
+            .build();
+
+    private final static List<PropertyDescriptor> propertyDescriptors;
+    private final static Set<Relationship> relationships;
+
+    private static final long TICKET_RENEWAL_PERIOD = 60000;
+
+    protected KerberosProperties kerberosProperties;
+
+    protected volatile HiveConfigurator hiveConfigurator = new HiveConfigurator();
+    protected volatile UserGroupInformation ugi;
+
+    protected final AtomicBoolean isInitialized = new AtomicBoolean(false);
+
+    protected HiveOptions options;
+    protected ExecutorService callTimeoutPool;
+    protected transient Timer heartBeatTimer;
+    protected AtomicBoolean sendHeartBeat = new AtomicBoolean(false);
+    protected Map<HiveEndPoint, HiveWriter> allWriters;
+
+
+    /*
+     * Will ensure that the list of property descriptors is build only once.
+     * Will also create a Set of relationships
+     */
+    static {
+        propertyDescriptors = new ArrayList<>();
+        propertyDescriptors.add(METASTORE_URI);
+        propertyDescriptors.add(HIVE_CONFIGURATION_RESOURCES);
+        propertyDescriptors.add(DB_NAME);
+        propertyDescriptors.add(TABLE_NAME);
+        propertyDescriptors.add(PARTITION_COLUMNS);
+        propertyDescriptors.add(AUTOCREATE_PARTITIONS);
+        propertyDescriptors.add(MAX_OPEN_CONNECTIONS);
+        propertyDescriptors.add(HEARTBEAT_INTERVAL);
+        propertyDescriptors.add(TXNS_PER_BATCH);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        _relationships.add(REL_RETRY);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        kerberosProperties = getKerberosProperties();
+        propertyDescriptors.add(kerberosProperties.getKerberosPrincipal());
+        propertyDescriptors.add(kerberosProperties.getKerberosKeytab());
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+
+    @OnScheduled
+    public void setup(final ProcessContext context) {
+        ComponentLog log = getLogger();
+
+        final String metastoreUri = context.getProperty(METASTORE_URI).getValue();
+        final String dbName = context.getProperty(DB_NAME).getValue();
```

Comment:
Would it make sense to support expression language for DB Name and Table Name? I suspect it would add a bit of complexity since you wouldn't be able to create the HiveOptions and login upfront and you'd have to do that each time through onTrigger and maintain some kind of cache of options, just throwing it out there as in idea